### ISSUE TITLE
Tag links are pointing to a nonSEF URL in multilanguage mode

### ISSUE THREAD

**Issue by hakanara** [joomla/joomla-cms#31530]
### Executive summary
Tag links on **Tags Popular module** and **articles** are pointing to a **nonSEF URL** in multilanguage mode like www.mysite.com/component/tags/tag/joomla.html even if there is a menu item that is set to "List all tags".

### Steps to reproduce the issue
- Setup a new MULTILANGUAGE web site by following these steps (Fresh setup for multilanguage site):
  https://docs.joomla.org/J3.x:Setup_a_Multilingual_Site/Installing_New_Language

- Create 2 "List All Tags" menu items for each language (In my case Turkish and English)
  (Where the created menu items lead to: index.php?option=com_tags&view=tags)

- In Global Configuration, say yes to these 3 settings:
  Search Engine Friendly URLs
  Use URL Rewriting
  Add Suffix to URL 

- Rename htaccess file on the server as .htaccess.

### Expected result
Tags on the "Tags Popular" module pointing to a menu like:
www.mysite.com/tr/myTurkishTagsAlias/joomla and 
www.mysite.com/en/myEnglishTagsAlias/joomla.
(Joomla is the only available tag after the setup.)

### Actual result
www.mysite.com/en/component/tags/tag/joomla.html
www.mysite.com/tr/component/tags/tag/joomla.html
(This issue is the same for blog post tags and the tags in "tags popular" module.)

_If there is only one tag menu item assigned to all languages, it works as I have expected above._

### System information (as much as possible)
Joomla 3.9.23 (Latest as of today)

### Additional comments
You can read the same issue here that was written many years ago:
https://forum.joomla.org/viewtopic.php?t=859288

It might be a router issue because the same issue arises on the blog view. The tags of the articles also lead to a link like component/tags/tag/joomla.html when there is more than one tags menu item that are assigned to different languages.

Thank you very much to all the members of Joomla team who are putting their love to this wonderful CMS.

**Comment by hakanara** [comment]
After making some more research, I realize that this conflict only arises when you set the language of "List all tags" menu item to a specific language. If the language of the menu item is set to "all" the links look OK. So also in this case, I couldn't assign two different alias to tags for two different languages.

**Comment by infograf768** [comment]
have you set Tags Options to use "Current"?
<img width="700" alt="Screen Shot 2020-12-05 at 10 35 24" src="https://user-images.githubusercontent.com/869724/101239164-af379c00-36e5-11eb-92b8-2d9174cd201f.png">

**Comment by hakanara** [comment]
Hello @infograf768 

Actually I saw it from one of your old posts on Joomla forum and had already tried changing the setting as "current". The links are still as I have described above. Like:
www.mysite.com/tr/component/tags/tag/joomla.html

This never happens if we create different aliases for blog or from any other component with different language selections. So this is interesting. I have been observing this for about a year on my site. This is not something new. According to the Joomla forum post from 2014, the same issue was still there then.

Thank you for your time. For thinking on it and answering me.

**Comment by alikon** [comment]
please test https://github.com/joomla/joomla-cms/pull/37721

**Comment by hakanara** [comment]
Hello @alikon . I have been thinking about this issue recently (which I have opened 1,5 years ago). Thank you very much for all the contributors. Somebody heard us. I can't test it right now but I would like to ask, will this fix be applied to Joomla 3.10 as well?

**Comment by simbus82** [comment]
In Joomla 3.10.10 this is really broken... The tags list in a category view generate nonSEF url.

@alikon @infograf768 
We have translated an IT site to EN and now all of our tagging system have broken URLs.
A non SEF url cause a lot of problems, like an impossible module assignement to a menu voice.

How we can solve?

We have duplicated and assigned to the right language every element (tags, menu voices and articles) and it's broken.
